demo-projects/meetup/site/pages/_app.js

```diff
@@ -9,37 +9,43 @@ import withApollo from '../lib/withApollo';
 import { AuthProvider } from '../lib/authetication';
 import StylesBase from '../primitives/StylesBase';
 
-const apiEndpoint = 'http://localhost:3000/api';
-
 class MyApp extends App {
   static async getInitialProps({ Component, ctx }) {
     let pageProps = {};
 
-    // We need to forward the request headers on the server.
-    const url = `${apiEndpoint}/session`;
-    const headers = ctx.req ? { cookie: ctx.req.headers.cookie } : undefined;
-
-    const { user } = await fetch(url, { headers }).then(res => res.json());
+    const data = await ctx.apolloClient.query({
+      query: gql`
+        query {
+          authenticatedUser {
+            id
+            name
+            isAdmin
+          }
+        }
+      `,
+      fetchPolicy: 'network-only',
+    });
 
     if (Component.getInitialProps) {
       pageProps = await Component.getInitialProps(ctx);
     }
-    return { pageProps, user };
+
+    return { pageProps, user: data.data ? data.data.authenticatedUser : undefined };
   }
 
   render() {
     const { Component, pageProps, apolloClient, user } = this.props;
     return (
       <ToastProvider>
         <Container>
-          <AuthProvider initialUserValue={user}>
-            <ApolloProvider client={apolloClient}>
+          <ApolloProvider client={apolloClient}>
+            <AuthProvider initialUserValue={user}>
```

Comment:
`AuthProvider` needs access to Apollo, so I pushed the `ApolloProvider` further up the stack.